backend/api/models.py

```
@@ -26,73 +35,99 @@ class User(AbstractBaseUser, PermissionsMixin):
+        choices=VISIBILITY_CHOICES,
+        default='private'
+    )
+    video_url = models.URLField(max_length=1024) 
+    thumbnail_url = models.URLField(max_length=1024, blank=True)
+    upload_date = models.DateTimeField(auto_now_add=True, db_index=True) 
     uploader = models.ForeignKey(User, on_delete=models.CASCADE, related_name='videos')
+
+  
+    views = models.PositiveIntegerField(default=0, validators=[MinValueValidator(0)])
+    likes = models.PositiveIntegerField(default=0, validators=[MinValueValidator(0)])
+    duration = models.CharField(max_length=50, blank=True, null=True) 
```

Ruff (0.8.2)

Avoid using `null=True` on string-based fields such as `CharField`

(DJ001)